src/main/java/uk/gov/hmcts/reform/roleassignment/controller/advice/RoleAssignmentControllerAdvice.java

```diff
@@ -13,22 +14,26 @@
 import javax.servlet.http.HttpServletRequest;
 
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.converter.HttpMessageConversionException;
+import org.springframework.http.converter.HttpMessageNotReadableException;
 import org.springframework.web.HttpMediaTypeNotAcceptableException;
 import org.springframework.web.bind.MethodArgumentNotValidException;
-import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import uk.gov.hmcts.reform.roleassignment.apihelper.Constants;
 import uk.gov.hmcts.reform.roleassignment.controller.advice.exception.BadRequestException;
 import uk.gov.hmcts.reform.roleassignment.controller.advice.exception.InvalidRequest;
 import uk.gov.hmcts.reform.roleassignment.controller.advice.exception.ResourceNotFoundException;
+import uk.gov.hmcts.reform.roleassignment.util.ValidationUtil;
 
 @Slf4j
-@ControllerAdvice(basePackages = "uk.gov.hmcts.reform.roleassignment")
+@RestControllerAdvice(basePackages = "uk.gov.hmcts.reform.roleassignment")
```

Comment:
This is the reason the exception handling for errors thrown by Libraries was not working.